pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -145,22 +148,64 @@ func (h *NetworkUtilsHandler) ChangeMacAddr(iface string) (net.HardwareAddr, err
 }
 
 func (h *NetworkUtilsHandler) StartDHCP(nic *VIF, serverAddr *netlink.Addr) {
+	nameservers, err := getNameserversFromPod()
+	if err != nil {
+		log.Log.Errorf("Failed to get DNS servers from resolv.ocnf: %v", err)
+		panic(err)
+	}
+
 	// panic in case the DHCP server failed during the vm creation
 	// but ignore dhcp errors when the vm is destroyed or shutting down
-	if err := DHCPServer(
+	if err = DHCPServer(
 		nic.MAC,
 		nic.IP.IP,
 		nic.IP.Mask,
 		api.DefaultBridgeName,
 		serverAddr.IP,
 		nic.Gateway,
-		net.ParseIP(guestDNS),
+		nameservers,
 	); err != nil {
 		log.Log.Errorf("failed to run DHCP: %v", err)
 		panic(err)
 	}
 }
 
+func getNameserversFromPod() ([][]byte, error) {
+	b, err := ioutil.ReadFile(resolvConf)
+	if err != nil {
+		return nil, err
+	}
+
+	return ParseNameservers(string(b))
+}
+
+func ParseNameservers(content string) ([][]byte, error) {
+	var nameservers [][]byte
+
+	re, err := regexp.Compile("([0-9]{1,3}.?){4}")
+	if err != nil {
+		return nameservers, err
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(content))
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "nameserver") {
+			nameserver := re.FindString(line)
+			if nameserver != "" {
+				nameservers = append(nameservers, net.ParseIP(nameserver).To4())
+			}
+		}
+	}
+
+	if err = scanner.Err(); err != nil {
+		return nameservers, err
+	}
+
```

Comment:
I think it would be useful to still have '8.8.8.8' as a sane default in the case that no nameservers are listed in the pods resolv.conf for some reason.

Reply:
Yes, I agree. I actually missed that. We should have it as a default.